atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -245,19 +251,34 @@ SweepOutcome runOnce() {
         return specificTableSweeper.getTxManager().runTaskWithRetry(
                 tx -> {
                     Optional<SweepProgress> progress = specificTableSweeper.getSweepProgressStore().loadProgress();
-                    if (progress.isPresent()) {
+                    SweepPriorityOverrideConfig overrideConfig = sweepPriorityOverrideConfig.get();
+                    if (progress.map(realProgress -> shouldContinueSweepingCurrentTable(realProgress, overrideConfig))
```

Comment:
If we stop sweeping a table because of an override then do we need to do any clean up of the SweepProgressStore?  Is it ok for the SweepProgressStore to have the interrupted table and the priority table in it?

Also, since we don't complete the table we don't write the final results to the SweepPriorityStore.  Does this have an impact on the priority calculation in subsequent runs?

Reply:
Based on answers to this there may be follow up work or the documentation may need to be updated with the implications of any interruption to a table being swept.

Reply:
Yes, agree that the implications of blacklisting/prioritizing tables could be better documented.

Sweep progress store:
We could clear the table, though I'm not sure there's a very significant benefit. Currently, the old entry for sweep progress will remain in the table; however, once the first batch of the new table is cleared it goes away. SweepProgressStore basically only stores 1 value - and supports PUT, GET and DELETE methods. This also means that after the blacklist/priority settings are removed, sweep won't return to where it left off. Will add some clarity in the docs.

Priority:
Currently we don't flush the stats to the sweep priority store, meaning that a non prioritized table will still retain its priority from before the sweep started. This likely means that the probability this table will be picked the next time for sweeping will increase, which seems fine. We could try to do something cleverer (e.g. by trying to scale down the number of writes and time since last sweep by some factor - possibly based on how far the sweep got assuming a uniform distribution across bytes), though I can't think of a good solution in general.

Reply:
Thanks for the details.  I don't think we need to make any code changes, just want us to be clear on the tradeoffs when using this.